solutions/devsprint-michel-ferreira-1/src/main/kotlin/com/devpass/challengehexagonal/resources/repository/database/client/TransactionSpringDataRepository.kt

```diff
@@ -13,4 +13,9 @@ interface TransactionSpringDataRepository : CrudRepository<TransactionEntity, Lo
         startDateTime: LocalDateTime,
         endDateTime: LocalDateTime,
     ): List<TransactionEntity>
+
```

Comment:
@RiellyCM nesse caso nós precisaremos da primeira transaction do dia.
Nosso SQL precisará ser mais ou menos assim:
Select * from transaction 
where account_id = ?
and transaction_date between ? and ?
limit 1

Para representar isso via Spring Data, precisamos que o método siga o seguinte padrão:
fun findTopByAccountIdAndTransactionDateBetweenOrderByTransactionDateAsc(
    accountId: Long,
    startDate: LocalDateTime,
    endDate: LocalDateTime
) : Transaction?

findTop -> buscará o primeiro resultado da consulta (limit 1)
by -> representa nossa cláusula 'Where'
accountId -> filtra pela accountId (account_id = ?)
AndTransactionDateBetween -> cria o filtro da data (and transaction_date between ? and ?)
OrderByTransactionDateAsc -> ordena nossa consulta pela transaction_date em ordem crescente

Vamos abordar isso no nosso encontro para ficar claro para todos :)